src/App.tsx

```diff
@@ -1,14 +1,73 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList';
+
+type FilterParams = {
+  query: string;
+};
+
+function filterPeople(people: Person[], { query }: FilterParams): Person[] {
+  let resultPeople = [...people];
+  const normalazidQuery = query.toLowerCase();
+
+  if (query) {
+    resultPeople = resultPeople.filter((person) => {
+      const normalazidName = person.name.toLowerCase();
+
+      return normalazidName.includes(normalazidQuery);
+    });
+  }
+
+  return resultPeople;
+}
 
 export const App: React.FC = () => {
```

Comment:
The task was to create a separate component `Autocomplete` and move most functionality there

Reply:
Yup, Implement the Autocomplete component